viewing_party/party.py

```diff
@@ -0,0 +1,188 @@
+    rating_numerator = 0
+    
+    for movie in user_data["watched"]:
+        rating_numerator += movie["rating"]
+
+    if len(user_data["watched"]) == 0:
+        return average_rating
+    else:
+        average_rating = (rating_numerator / movies_watched)
+        return average_rating
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    genre_dict = {}
+    most_frequent = ""
+    frequency = 0
+    
+    if len(user_data["watched"]) == 0:
+        return None
+    
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+    
+    for genre in genre_list:
+        if genre in genre_dict:
+            genre_dict[genre] += 1
+        else:
+            genre_dict[genre] = 1
+    
+    for key, value in genre_dict.items():
+        if value > frequency:
+            frequency = value
+            most_frequent = key
+    
+    return most_frequent
+
+# wave 3
+def get_unique_watched(user_data):
+    friend_movie_list = []
+    user_movie_list = []
+    unique_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movie_list.append(movie["title"])
+
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    unique_watched = set(user_movie_list) - set(friend_movie_list)
+    
+    for movie in unique_watched:
+        unique_list.append({"title": movie})
+    
+    return unique_list
+
+def get_friends_unique_watched(user_data):
+    friend_movie_list = []
+    user_movie_list = []
+    unique_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movie_list.append(movie["title"])
+    
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    unique_watched = set(friend_movie_list) - set(user_movie_list)
+    
+    for movie in unique_watched:
+        unique_list.append({"title": movie})
+    
+    return unique_list
+
+# wave 4
+def get_available_recs(user_data):
+    possible_recommendations = []
+    host_list = []
+    user_movie_list = []
+    final_list = []
+    friend_host_list = []
+
+    for movie in user_data["watched"]:
+        user_movie_list.append(movie["title"])
+
+    for host in user_data["subscriptions"]:
+        host_list.append(host)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["host"] in host_list:
+                possible_recommendations.append(movie["title"])
+                friend_host_list.append(movie["host"])
+
+    recommendations = set(possible_recommendations) - set(user_movie_list)
+
+    if recommendations == False:
+        return None
+
+    for item in recommendations:
+        i = possible_recommendations.index(item)
+        final_list.append({"title": possible_recommendations[i], "host": friend_host_list[i]})
+    
+    return final_list
+
+# wave 5
+def get_new_rec_by_genre(user_data):
```

Comment:
This function is very similar to to `get_available_recs` except that the criteria for which movies to take from the friends lists is predicated on the user's favorite genre, rather than the subscriptions. Similar comments apply here, including the alternative approach of reworking wave 3 to allow using it here, then filtering the the unique friends watched movies to those which match the user's favorite genre.